voltdbclient/conn.go

```diff
@@ -377,68 +391,40 @@ func readLoginResponse(reader io.Reader) (*connectionData, error) {
 }
 
 type VoltAsyncResponse struct {
-	conn    VoltConn
-	han     int64
-	ch      <-chan VoltResponse
-	isQuery bool
-	result  driver.Result
-	rows    driver.Rows
-	err     error
-	active  bool
+	conn     VoltConn
+	han      int64
+	ch       <-chan VoltResponse
+	isQuery  bool
+	result   driver.Result
+	rows     driver.Rows
+	resCons  resultConsumer
+	rowsCons rowsConsumer
 }
 
-func newVoltAsyncResponse(conn VoltConn, han int64, ch <-chan VoltResponse, isQuery bool) *VoltAsyncResponse {
+type rowsConsumer func(driver.Rows, error)
+
+type resultConsumer func(driver.Result, error)
+
```

Comment:
Maybe we can make the callback function an interface type with a 
clientCallback( clientReposne) method? (the same way we define ProcedureCallback for java client).

Then we can pass more general callback functions.